File: TripKitAndroid/src/main/java/com/skedgo/tripkit/LocationInfoDetails.java
package com.skedgo.tripkit;

import androidx.annotation.Nullable;

import com.google.gson.annotations.JsonAdapter;

import org.immutables.gson.Gson;
import org.immutables.value.Value;

@Gson.TypeAdapters
@Value.Immutable
@Value.Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(GsonAdaptersLocationInfoDetails.class)
public abstract class LocationInfoDetails {
  @Nullable public abstract String w3w();
  @Nullable public abstract String w3wInfoURL();
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/LocationInfoDetails.kt
package com.skedgo.tripkit

import com.google.gson.annotations.JsonAdapter
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@TypeAdapters
@Immutable
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersLocationInfoDetails::class
)
abstract class LocationInfoDetails {
    abstract fun w3w(): String?

    abstract fun w3wInfoURL(): String?
}